src/foam/nanos/crunch/lite/RequirementsRuleAction.js

```diff
@@ -0,0 +1,73 @@
+foam.CLASS({
+  package: 'foam.nanos.crunch.lite',
+  name: 'RequirementsRuleAction',
+
+  documentation: `
+    Rule action that adds payloads to a Capable object.
+    Can only be used on objects implementing Capable.
+  `,
+
+  implements: [
+    'foam.nanos.ruler.RuleAction'
+  ],
+
+  javaImports: [
+    'foam.core.ContextAgent',
+    'foam.core.X',
+    'foam.dao.DAO',
+    'foam.nanos.crunch.CapabilityJunctionPayload',
+    'foam.nanos.alarming.Alarm',
+    'foam.nanos.alarming.AlarmReason',
+  ],
+
+  properties: [
+    {
+      class: 'StringArray',
+      name: 'requirements'
+    },
+    {
+      class: 'StringArray',
+      name: 'available',
+      documentation: 'Adds a payload but does not add to capabilityIds'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'applyAction',
+      javaCode: `
+        agency.submit(x, new ContextAgent() {
+          @Override
+          public void execute(X x) {
+            if ( ! (obj instanceof Capable) ) {
```

Comment:
hmm - you are limited to Capable properties if you do this. You know you are getting a TPI obj. You should cast it accordingly.